### PR TITLE
Contains tasks 2,3 and 5 of the 3rd theme.

### PR DESCRIPTION
**Pull request by iliyanmotovski**
2) Scan until reach separator and write to file
3) Reverse the contents of a file using buffer
5) Use gobs package to write go object to a file, then read it

### CODE REVIEW

File: scanuntilsepapp/main.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+var fileName = flag.String("fileName", "DefaultFile.txt", "target file")
+
+func main() {
+	flag.Parse()
+
+	//ScanUntilReachSep task
+	fmt.Println("Enter some input: ")
+	result := bytes.NewBufferString(stream.ScanUntilReachSep(os.Stdin, "."))
+	err := stream.Write(*fileName, result)
```
Comment (mgenov) [inline scanuntilsepapp/main.go]: Doesn't look correct as I'm not able to see how stream is imported.

File: scanuntilsepapp/main.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+	if err != nil {
+		err = stream.Write("DefaultFileName.txt", result)
+		if err != nil {
+			log.Fatal(err)
```
Comment (mgenov) [inline scanuntilsepapp/main.go]: log.Fatalf("could not write content to DefaultFileName.txt due: %v", err)

File: inputoutputpack/scanuntilsep.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package stream
```
Comment (mgenov) [inline inputoutputpack/scanuntilsep.go]: package is stream but it stays in inputoutputpack which doesn't look good.

File: inputoutputpack/bufferreverse.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
+package stream
```
Comment (mgenov) [inline inputoutputpack/bufferreverse.go]: package is stream but file stays in the inputoutputpack folder.

File: inputoutputpack/scanuntilsep.go (hunk 3 above)
Comment (iliyanmotovski) [inline inputoutputpack/scanuntilsep.go]: I will modify the folder name. Notice taken that package name must be the same as the folder within it is located.

File: scanuntilsepapp/main.go (hunk 2 above)
Comment (iliyanmotovski) [inline scanuntilsepapp/main.go]: Will implement your suggestion in the solution.

File: scanuntilsepapp/main.go (hunk 1 above)
Comment (iliyanmotovski) [inline scanuntilsepapp/main.go]: Notice taken.

**Comment by iliyanmotovski** [comment]
All changes requested are implemented.

**Pull request closed without merging**